tests/components/abode/test_light.py

```diff
@@ -0,0 +1,135 @@
+"""Tests for the Abode light device."""
+from homeassistant.components.light import DOMAIN as LIGHT_DOMAIN
+
+from .common import setup_platform
+
+
+async def test_entity_registry(hass, requests_mock):
+    """Tests that the devices are registered in the entity registry."""
+    await setup_platform(hass, LIGHT_DOMAIN)
+    entity_registry = await hass.helpers.entity_registry.async_get_registry()
+
+    entry = entity_registry.async_get("light.living_room_lamp")
+    assert entry.unique_id == "741385f4388b2637df4c6b398fe50581"
+
+
+async def test_attributes(hass, requests_mock):
+    """Test the light attributes are correct."""
+    await setup_platform(hass, LIGHT_DOMAIN)
+
+    state = hass.states.get("light.living_room_lamp")
+    assert state.state == "on"
+    assert state.attributes.get("brightness") == 204
+    assert state.attributes.get("rgb_color") == (0, 63, 255)
+    assert state.attributes.get("color_temp") == 280
+    assert state.attributes.get("device_id") == "ZB:db5b1a"
+    assert not state.attributes.get("battery_low")
+    assert not state.attributes.get("no_response")
+    assert state.attributes.get("device_type") == "RGB Dimmer"
+    assert state.attributes.get("friendly_name") == "Living Room Lamp"
+    assert state.attributes.get("supported_features") == 19
+
+
+async def test_switch_off(hass, requests_mock):
+    """Test the light can be turned off."""
+    await setup_platform(hass, LIGHT_DOMAIN)
+    requests_mock.put(
+        "https://my.goabode.com/api/v1/control/light/ZB:db5b1a",
+        json={"id": "ZB:db5b1a", "status": "0"},
+    )
+
+    assert await hass.services.async_call(
+        "light", "turn_off", {"entity_id": "light.living_room_lamp"}, blocking=True
+    )
+    await hass.async_block_till_done()
+    # Manually call the callback
+    device = hass.data["abode"].abode.get_device("ZB:db5b1a")
+    # pylint: disable=W0212
+    hass.data["abode"].abode.events._device_callbacks["ZB:db5b1a"][0](device)
```

Comment:
I'm trying to manually call a callback, but this doesn't seem to be working. I'm not quite sure why as this is my understanding of the flow:

When a device is added in Home Assistant during the setup, `add_device_callback` is called in [abodepy.event_controller.py](https://github.com/MisterWil/abodepy/blob/c4108eccea28d332926112b8f129ffe1b4b96ae8/abodepy/event_controller.py#L49) which appends the callbacks to the `self._device_callbacks` dict. The callback being appended is the `_update_callback` method in the `AbodeDevice` class in the [abode integration](https://github.com/home-assistant/home-assistant/blob/f3a8196fb5815fc2abbcc6f4f2f4ab0e0758f067/homeassistant/components/abode/__init__.py#L350). When an event is received from Abode's web socket server, it should simply be executing the `_update_callback`.

Reply:
I've given up on this approach. I'm now just mocking the abodepy method being called and testing if it's being called. I'll have to add separate tests later to test when event updates are received from abodepy.